src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangRecordProcessor.java

```diff
@@ -136,6 +141,24 @@ public void shutdown(ShutdownInput shutdownInput) {
         }
     }
 
+    @Override
+    public void shutdownRequested(IRecordProcessorCheckpointer checkpointer) {
+        LOG.info("Shutdown is requested.");
+        if (!initialized) {
+            LOG.info("Record processor was not initialized so no need to initiate a final checkpoint.");
+            return;
+        }
+        try {
+            LOG.info("Requesting a checkpoint on shutdown notification.");
+            checkpointer.checkpoint();
+        } catch (InvalidStateException e) {
+            LOG.error("Checkpoint triggered during shutdown encountered InvalidStateException: " + e.toString());
```

Comment:
Java will automatically call toString during string concatenation.  Additionally for exceptions it's common to pass the exception as the second argument to the logger call.  This will allow the logger to emit a stack trace as well e.g.
``` java
LOG.error("Checkpoint triggered during shutdown encountered InvalidStateException: " + e, e);
```